Deeploy/Targets/Generic/TypeCheckers.py

```
@@ -2,145 +2,84 @@
-        super().__init__(input_types, output_types)
+        maxNLevel = max(i.nLevels for i in inputs)
 
-    def _inferNumLevels(self, inputs: List[VariableBuffer],
-                        operatorRepresentation: OperatorRepresentation) -> List[int]:
-        return [inputs[0].nLevels]
+        return [maxNLevel]
 
     def _inferSignedness(self, inputs: List[VariableBuffer],
-                         operatorRepresentation: OperatorRepresentation) -> List[bool]:
+                         operatorRepresentation: OperatorRepresentation) -> Optional[List[bool]]:
+        assert (all([_inp._signed == True for _inp in inputs]) or all(
```

Ruff (0.15.20)

[error] Avoid equality comparisons to `True`; use `_inp._signed:` for truth checks

Replace with `_inp._signed`

(E712)